libqtile/core/manager.py

```diff
@@ -59,61 +57,67 @@
 from libqtile.widget.base import _Widget
 
 
-def handle_exception(loop, context):
-    if "exception" in context:
-        logger.error(context["exception"], exc_info=True)
-    else:
-        logger.error("exception in event loop: %s", context)
-
-
 class Qtile(CommandObject):
     """This object is the `root` of the command graph"""
     def __init__(
         self,
         kore,
         config,
-        eventloop,
         no_spawn=False,
         state=None
     ):
-        self._restart = False
+        self.core = kore
         self.no_spawn = no_spawn
+        self._state = state
 
-        self.mouse_position = (0, 0)
+        self._stopped_event = None
+        self.should_restart = False
 
-        self.core = kore
-        self.config = config
-        libqtile.init(self)
+        self._drag = None
+        self.mouse_map = None
+        self.mouse_position = (0, 0)
 
         self.windows_map = {}
         self.widgets_map = {}
         self.groups_map = {}
         self.groups = []
+        self.dgroups = None
+
         self.keys_map = {}
         self.current_chord = False
-
         self.numlock_mask, self.valid_mask = self.core.masks
 
+        self.current_screen = None
+        self.screens = []
+
+        libqtile.init(self)
+
+        self._event_loop = None
+        self.server = IPCCommandServer(self)
+        self.config = config
+        # config can potentially be replaced by the default config
+        self.config = self.load_config(config)
+
+    def load_config(self, config):
         try:
-            self.config.load()
+            config.load()
         except Exception as e:
             logger.exception('Error while reading config file (%s)', e)
-            self.config = confreader.Config()
+            config = confreader.Config()
             from libqtile.widget import TextBox
-            widgets = self.config.screens[0].bottom.widgets
+            widgets = config.screens[0].bottom.widgets
             widgets.insert(0, TextBox('Config Err!'))
 
-        self.core.wmname = getattr(self.config, "wmname", "qtile")
+        self.core.wmname = getattr(config, "wmname", "qtile")
```

Comment:
I thought that you wanted to replace `self.config` by `config` everywhere to make the function pure, but you still left side effects behind, so what's the point?